src/loaders/lottie/tvgLottieBuilder.cpp

```diff
@@ -710,16 +715,76 @@ static void _updateStar(LottieGroup* parent, LottiePolyStar* star, Matrix* trans
                 mathTransform(transform, &in3);
                 mathTransform(transform, &in4);
             }
-            merging->cubicTo(in2.x, in2.y, in3.x, in3.y, in4.x, in4.y);
+            shape->cubicTo(in2.x, in2.y, in3.x, in3.y, in4.x, in4.y);
         } else {
             Point in = {x, y};
             if (transform) mathTransform(transform, &in);
-            merging->lineTo(in.x, in.y);
+            shape->lineTo(in.x, in.y);
         }
         angle += dTheta * direction;
         longSegment = !longSegment;
     }
-    merging->close();
+    shape->close();
+
+    if (applyExtRoundness) {
+        auto cmdCnt = shape->pathCommands(nullptr);
+        const Point *pts = nullptr;
+        auto ptsCnt = shape->pathCoords(&pts);
+
+        auto len = mathLength(pts[1] - pts[2]);
+        auto r = len > 0.0f ? ROUNDED_POLYSTAR_MAGIC_NUMBER * mathMin(len * 0.5f, roundness) / len : 0.0f;
+
+        if (hasRoundness) {
```

Comment:
You might wanna double-check, we can skip round corner handling in this case—rounding a star—because we do not apply rounded corners to curved paths, only to straight lines.

Reply:
if in a bezier ```start == crtl1``` or```ctrl2 == end``` then roundness should be applied

|this branch | AE|
|------------|----|
|<img width="356" alt="Zrzut ekranu 2024-05-13 o 13 11 40" src="https://github.com/thorvg/thorvg/assets/67589014/4ece0b11-3778-4f70-8e2a-8be14cf586e4">|<img width="395" alt="Zrzut ekranu 2024-05-13 o 13 11 45" src="https://github.com/thorvg/thorvg/assets/67589014/dc1209d3-a14d-4f8a-a9ca-7b4627f07f35">|
|<img width="383" alt="Zrzut ekranu 2024-05-13 o 13 12 34" src="https://github.com/thorvg/thorvg/assets/67589014/73e35a38-2008-4990-b7e6-ab425df5898b">|<img width="443" alt="Zrzut ekranu 2024-05-13 o 13 12 40" src="https://github.com/thorvg/thorvg/assets/67589014/726435a0-8172-4b23-98d0-3215726a7130">|
|<img width="316" alt="Zrzut ekranu 2024-05-13 o 13 12 02" src="https://github.com/thorvg/thorvg/assets/67589014/dae73396-5d11-4e5c-88ff-ed207be9b2bb">|<img width="434" alt="Zrzut ekranu 2024-05-13 o 13 12 11" src="https://github.com/thorvg/thorvg/assets/67589014/7d0a5646-c22f-4f05-80e0-7bc393b7fbdd">|